src/platforms/web/runtime/directives/show.js

```diff
@@ -17,22 +17,24 @@ export default {
     if (value && transition && transition.appear && !isIE9) {
       enter(vnode)
     }
-    el.style.display = value ? '' : 'none'
+    const originalDisplay = el.style.display
+    el.style.display = value ? originalDisplay : 'none'
+    el.dataset.__vOriginalDisplay = originalDisplay
```

Comment:
@yyx990803 Wow, you're absolutely right - I can't believe I didn't know this! How should we handle this with the Flow types? Declare a custom type that extends `HTMLElement` and allows `__vOriginalDisplay` as a string?